Makefile

```
@@ -12,7 +12,442 @@
+
+TIMEOUT := $(shell command -v timeout || command -v gtimeout)
+
+# Define Docker related variables. Releases should modify and double check these vars.
+export GCP_PROJECT ?= $(shell gcloud config get-value project)
+REGISTRY ?= gcr.io/$(GCP_PROJECT)
+STAGING_REGISTRY := gcr.io/k8s-staging-cluster-api-equinix-metal
+PROD_REGISTRY := us.gcr.io/k8s-artifacts-prod/cluster-api-equinix-metal
+IMAGE_NAME ?= cluster-api-packet-controller
+export CONTROLLER_IMG ?= $(REGISTRY)/$(IMAGE_NAME)
+export TAG ?= dev
+export ARCH ?= amd64
+ALL_ARCH = amd64 arm64 ppc64le s390x
+
+# Allow overriding manifest generation destination directory
+MANIFEST_ROOT ?= config
+CRD_ROOT ?= $(MANIFEST_ROOT)/crd/bases
+WEBHOOK_ROOT ?= $(MANIFEST_ROOT)/webhook
+RBAC_ROOT ?= $(MANIFEST_ROOT)/rbac
+
+# Allow overriding the imagePullPolicy
+PULL_POLICY ?= Always
+
+# Hosts running SELinux need :z added to volume mounts
+SELINUX_ENABLED := $(shell cat /sys/fs/selinux/enforce 2> /dev/null || echo 0)
+
+ifeq ($(SELINUX_ENABLED),1)
+  DOCKER_VOL_OPTS?=:z
+endif
+
+# Build time versioning details.
+LDFLAGS := $(shell hack/version.sh)
+
+GOLANG_VERSION := 1.16.6
+
+## --------------------------------------
+## Help
+## --------------------------------------
+
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+## --------------------------------------
+## Testing
+## --------------------------------------
+
+.PHONY: test
+test: ## Run tests
+	source ./scripts/fetch_ext_bins.sh; fetch_tools; setup_envs; go test -v ./... -coverprofile cover.out
+
+# Allow overriding the e2e configurations
+GINKGO_FOCUS ?= Workload cluster creation
+GINKGO_NODES ?= 3
+GINKGO_NOCOLOR ?= false
+ARTIFACTS ?= $(ROOT_DIR)/_artifacts
+SKIP_CLEANUP ?= false
+SKIP_CREATE_MGMT_CLUSTER ?= false
+E2E_DATA_DIR ?= $(TEST_E2E_DIR)/data
+KUBETEST_CONF_PATH ?= $(abspath $(E2E_DATA_DIR)/kubetest/conformance.yaml)
+
+.PHONY: test-e2e-capi
+test-e2e-capi: $(ENVSUBST) $(KUBECTL) $(GINKGO) e2e-image e2e-test-templates ## Run the end-to-end tests
+	$(ENVSUBST) < $(E2E_CONF_FILE) > $(E2E_CONF_FILE_ENVSUBST) && \
+	time $(GINKGO) -v -trace -progress -v -tags=e2e -focus='Running the Cluster API E2E tests' \
+		-nodes=$(GINKGO_NODES) --noColor=$(GINKGO_NOCOLOR) \
+		--flakeAttempts=2 ./test/e2e -- \
+		-e2e.artifacts-folder="$(ARTIFACTS)" \
+		-e2e.config="$(E2E_CONF_FILE_ENVSUBST)" \
+		-e2e.skip-resource-cleanup=$(SKIP_CLEANUP) \
+		-e2e.use-existing-cluster=$(SKIP_CREATE_MGMT_CLUSTER)
+
+.PHONY: test-e2e-capi-smoketest
+test-e2e-capi-smoketest: $(ENVSUBST) $(KUBECTL) $(GINKGO) e2e-image e2e-test-templates ## Run the end-to-end tests
+	$(ENVSUBST) < $(E2E_CONF_FILE) > $(E2E_CONF_FILE_ENVSUBST) && \
+	time $(GINKGO) -v -trace -progress -v -tags=e2e -focus='\[Smoke Test\] \
```

missing `'` in the end
```suggestion
	time $(GINKGO) -v -trace -progress -v -tags=e2e -focus='\[Smoke Test\]' \
```